lib/repl.js

```diff
@@ -222,12 +127,24 @@ function REPLServer(prompt,
   eval_ = eval_ || defaultEval;
 
   function defaultEval(code, context, file, cb) {
-    var err, result, retry = false, input = code, wrappedErr;
+    var err, result, wrappedErr;
+    var retry = false;
+    var wrappedCmd = false;
+    var input = code;
+
+    if (/^\s*\{/.test(code) && /\}\s*$/.test(code)) {
+      // It's confusing for `{ a : 1 }` to be interpreted as a block
+      // statement rather than an object literal.  So, we first try
+      // to wrap it in parentheses, so that it will be interpreted as
+      // an expression.
+      code = '(' + code.trim() + ')\n';
+      wrappedCmd = true;
+    }
+
     // first, create the Script object to check the syntax
     while (true) {
       try {
-        if (!/^\s*$/.test(code) &&
```

Comment:
It's been a while, but IIRC correctly you can never get here with just whitespace because there's a check on the trimmed code before trying to eval. It made sense to move it there since it's applicable to all languages, not just the default eval.